cmd/provider/remove.go

```diff
@@ -44,25 +44,25 @@ specified, they key is simply calculated as the SHA_256 hash of the given path.`
 )
 
 func beforeRemoveCar(cctx *cli.Context) error {
-	keyFlagSet := cctx.IsSet(keyFlag.Name)
-	carPathFlagSet := cctx.IsSet(carPathFlag.Name)
-
-	if keyFlagSet && carPathFlagSet {
-		return fmt.Errorf("only one of %s or %s must be set", keyFlag.Name, carPathFlag.Name)
-	}
-	if !keyFlagSet && !carPathFlagSet {
-		return fmt.Errorf("either %s or %s must be set", keyFlag.Name, carPathFlag.Name)
+	if !cctx.IsSet(keyFlag.Name) {
+		if !cctx.IsSet(optionalCarPathFlag.Name) {
+			return fmt.Errorf("either %s or %s must be set", keyFlag.Name, optionalCarPathFlag.Name)
+		}
+		fmt.Println("carPathFlagValue:", optionalCarPathFlagValue)
+		h := sha256.New()
+		h.Write([]byte(optionalCarPathFlagValue))
+		removeCarKey = h.Sum(nil)
+		return nil
 	}
 
-	if keyFlagSet {
-		decoded, err := base64.StdEncoding.DecodeString(keyFlagValue)
-		if err != nil {
-			return errors.New("key is not a valid base64 encoded string")
-		}
-		removeCarKey = decoded
-	} else {
-		removeCarKey = sha256.New().Sum([]byte(carPathFlagValue))
+	if cctx.IsSet(optionalCarPathFlag.Name) {
+		return fmt.Errorf("only one of %s or %s must be set", keyFlag.Name, optionalCarPathFlag.Name)
+	}
+	decoded, err := base64.StdEncoding.DecodeString(keyFlagValue)
```

Comment:
can we have usage for this flag indicate it should be base64 encoded when passed in?

Reply:
The usage for `keyFlag` indicates base64:
```go
Usage:       "Base64 encoded lookup key to associate with imported CAR."
```
https://github.com/filecoin-project/index-provider/blob/main/cmd/provider/flags.go#L122

The `optionalCarPathFlag` is a path so should not be base64 encoded.